x/gov/client/cli/tx.go

```diff
@@ -155,29 +169,58 @@ func parseSubmitProposalFlags() (*proposal, error) {
 }
 
 // GetCmdDeposit implements depositing tokens for an active proposal.
-func GetCmdDeposit(cdc *codec.Codec) *cobra.Command {
+func GetCmdDeposit(queryRoute string, cdc *codec.Codec) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "deposit",
+		Use:   "deposit [proposal-id] [deposit]",
+		Args:  cobra.ExactArgs(2),
 		Short: "Deposit tokens for activing proposal",
+		Long: strings.TrimSpace(`
+Submit a deposit for an acive proposal. You can find the proposal-id by running gaiacli query gov proposals:
+
+$ gaiacli tx gov deposit 1 10STAKE --from mykey
+`),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			txBldr := authtxb.NewTxBuilderFromCLI().WithCodec(cdc)
 			cliCtx := context.NewCLIContext().
 				WithCodec(cdc).
 				WithAccountDecoder(cdc)
 
-			depositorAddr, err := cliCtx.GetFromAddress()
+			// validate that the proposal id is a uint
+			proposalID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("proposal-id %s not a valid int, please input a valid proposal-id", args[0])
+			}
+
+			// check to see if the proposal is in the store
+			_, err = queryProposal(proposalID, cliCtx, cdc, queryRoute)
+			if err != nil {
+				return fmt.Errorf("Failed to fetch proposal-id %d: %s", proposalID, err)
+			}
+
+			// Get from address
+			from, err := cliCtx.GetFromAddress()
 			if err != nil {
 				return err
 			}
 
-			proposalID := uint64(viper.GetInt64(flagProposalID))
+			// Fetch associated account
+			account, err := cliCtx.GetAccount(from)
+			if err != nil {
+				return err
+			}
 
-			amount, err := sdk.ParseCoins(viper.GetString(flagDeposit))
+			// Get amount of coins
+			amount, err := sdk.ParseCoins(args[1])
 			if err != nil {
 				return err
 			}
 
-			msg := gov.NewMsgDeposit(depositorAddr, proposalID, amount)
+			// ensure account has enough coins
```

Comment:
Same comment as above (we should simulate instead).

Reply:
Same comment as above